From 380fe7d4548a99bfcfc1594b6f0b3dd2369978f1 Mon Sep 17 00:00:00 2001
From: Hans Verkuil <hans.verkuil@cisco.com>
Date: Sun, 1 Apr 2018 12:10:33 +0200
Subject: [PATCH] libv4lconvert: fix compiler warning

  CC       control/libv4lconvert_la-libv4lcontrol.lo
control/libv4lcontrol.c: In function ‘v4lcontrol_get_usb_info’:
control/libv4lcontrol.c:391:13: warning: In the GNU C Library, "minor" is defined
 by <sys/sysmacros.h>. For historical compatibility, it is
 currently defined by <sys/types.h> as well, but we plan to
 remove this soon. To use "minor", include <sys/sysmacros.h>
 directly. If you did not intend to use a system-defined macro
 "minor", you should undefine it after including <sys/types.h>.
       c == '\n' && minor_dev == minor(st.st_rdev))
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Dropped the '#if defined(MAJOR_IN_SYSMACROS)' to ensure that sysmacros.h
is always included. It seems that #if is no longer needed. At least,
nobody complained when I removed it in other places earlier :-)

Signed-off-by: Hans Verkuil <hans.verkuil@cisco.com>
Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: commit 380fe7d4548a9

 lib/libv4lconvert/control/libv4lcontrol.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/libv4lconvert/control/libv4lcontrol.c b/lib/libv4lconvert/control/libv4lcontrol.c
index 1e784eda8a92..59f28b137b98 100644
--- a/lib/libv4lconvert/control/libv4lcontrol.c
+++ b/lib/libv4lconvert/control/libv4lcontrol.c
@@ -20,9 +20,7 @@
  */
 
 #include <sys/types.h>
-#if defined(MAJOR_IN_SYSMACROS)
 #include <sys/sysmacros.h>
-#endif
 #include <sys/mman.h>
 #include <fcntl.h>
 #include <sys/stat.h>
-- 
2.18.0

